packages/react/src/2016-12-01/button.js

```diff
@@ -19,14 +19,39 @@ require('@orion-ui/components/lib/2016-12-01/button');
 
 const React = require('react');
 const { Skins } = require('@orion-ui/style/lib/2016-12-01');
+const ButtonState = require('@orion-ui/components/lib/2016-12-01/button-state.js');
 
 const PropTypes = React.PropTypes;
 
-function Button(props) {
-  return (
-    <orion-button onClick={props.onClick} color={props.color} background={props.background}>
-      {props.children}
-    </orion-button>);
+class Button extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = ButtonState.create({ disabled: props.disabled });
```

Comment:
You are setting disabled prop in the constructor. What happens when disabled is changed at runtime?